pkg/endpoint/policy.go

```diff
@@ -475,7 +475,8 @@ func (e *Endpoint) updateRegenerationStatistics(ctx *regenerationContext, err er
 	stats.SendMetrics()
 
 	fields := logrus.Fields{
-		logfields.Reason: ctx.Reason,
+		logfields.Reason:     ctx.Reason,
+		logfields.K8sPodName: e.GetK8sNamespaceAndPodName(),
```

Comment:
It looks like this field should already be included in the logs below as part of the `scopedLog := e.getLogger().WithFields(fields)`, since the `Endpoint.getLogger()` function will be invoked. I would expect that the endpoint's logger was already updated with this info via `Endpoint.UpdateLogger()`.

I wonder if there is some other earlier location that is missing the `UpdateLogger()` call that should prepare this properly?

Reply:
These are good points, let me do some testing and get back to you on this.